chore: Add v0.15.0 CHANGELOG notes. (#9220)
ipfs/kubo: docs/changelogs/v0.15.md

# Kubo changelog v0.15

## v0.15.0

### Overview

> Below is an outline of all that is in this release, so you get a sense of all that's included.

Author: It is the same as in 0.14: https://github.com/ipfs/kubo/blob/master/docs/changelogs/v0.14.md#overview

> Below is an outline of all that is in this release, so you get a sense of all that's included.

Author: and 0.13

> Below is an outline of all that is in this release, so you get a sense of all that's included.

Reviewer: ok nvm keep it

- [Kubo changelog v0.15](#kubo-changelog-v015)
  - [v0.15.0](#v0150)
    - [Overview](#overview)
    - [🔦 Highlights](#-highlights)
      - [#️⃣ Blake support](#️⃣-blake-support)
      - [💉 Fx Options plugin](#-fx-options-plugin)
    - [Changelog](#changelog)
    - [Contributors](#contributors)

### 🔦 Highlights

This is a release mainly with bugfixing and library updates. 
We are improving release speed and cadence trying to have a new release every 5 weeks.

#### #️⃣ Blake support

> You can now use `blake3` as a valid hash function: `ipfs block put --mhtype=blake3`.
> It uses a 32 bytes default size.
> To use a different hash length, set `mhlen`: `ipfs block put --mhtype=blake3 --mhlen=64`

Reviewer: ```suggestion
You can now use `blake3` as a valid hash function:
- `ipfs block put --mhtype=blake3`
- `ipfs add --hash=blake3`
It uses a 32 bytes default size.
And verify up to 128 bytes.
Because `blake3` is variable output hash function, you can use a different digest length, set `mhlen`: `ipfs block put --mhtype=blake3 --mhlen=64`, `ipfs add` doesn't have this option yet.
```

  - add LocalAddr and RemoteAddr functions to http3.StreamCreator (#3384) ([lucas-clemente/quic-go#3384](https://github.com/lucas-clemente/quic-go/pull/3384))
  - extend the HTTP/3 API for WebTransport support ([lucas-clemente/quic-go#3362](https://github.com/lucas-clemente/quic-go/pull/3362))
  - remove unneeded network from custom dial function used in HTTP/3 (#3368) ([lucas-clemente/quic-go#3368](https://github.com/lucas-clemente/quic-go/pull/3368))
- github.com/multiformats/go-multiaddr (v0.5.0 -> v0.6.0):
  - release v0.6.0 ([multiformats/go-multiaddr#178](https://github.com/multiformats/go-multiaddr/pull/178))
  - add WebTransport multiaddr components ([multiformats/go-multiaddr#176](https://github.com/multiformats/go-multiaddr/pull/176))
  - add ipcidr support (#177) ([multiformats/go-multiaddr#177](https://github.com/multiformats/go-multiaddr/pull/177))
  - add a Contains function (#172) ([multiformats/go-multiaddr#172](https://github.com/multiformats/go-multiaddr/pull/172))
- github.com/multiformats/go-multibase (v0.1.0 -> v0.1.1):
  - chore: release version 0.1.1

>   - fix: add new emoji codepoint for Base256Emoji 🐉

Reviewer: This should have been in `v0.14.0` seems like I forgot to bump to `v0.1.1`, anyway hopefully this doesn't break anyone.

# Kubo changelog v0.15

## v0.15.0

### Overview

Below is an outline of all that is in this release, so you get a sense of all that's included.

- [Kubo changelog v0.15](#kubo-changelog-v015)
  - [v0.15.0](#v0150)
    - [Overview](#overview)
    - [🔦 Highlights](#-highlights)
      - [#️⃣ Blake support](#️⃣-blake-support)
      - [💉 Fx Options plugin](#-fx-options-plugin)
    - [Changelog](#changelog)
    - [Contributors](#contributors)

### 🔦 Highlights

This is a release mainly with bugfixing and library updates. 
We are improving release speed and cadence trying to have a new release every 5 weeks.

#### #️⃣ Blake support

You can now use `blake3` as a valid hash function: `ipfs block put --mhtype=blake3`.
It uses a 32 bytes default size.

> To use other hash length set `mhlen`: `ipfs block put --mhtype=blake3 --mhlen=64`

Reviewer 2: ```suggestion
To use a different hash length, set `mhlen`: `ipfs block put --mhtype=blake3 --mhlen=64`
```

#### 💉 Fx Options plugin

This adds a plugin interface that lets the plugin modify the fx options that are passed to fx when the app is initialized. 

> This means plugins can inject their own implementations of IPFS interfaces.

Reviewer 2: ```suggestion
This means plugins can inject their own implementations of Kubo interfaces.
```

> This enables granular customization of go-ipfs behavior by plugins, such as:

Reviewer 2: ```suggestion
This enables granular customization of Kubo behavior by plugins, such as:
```

- Bitswap with custom filters (e.g. for CID blocking)
- Custom interface implementations such as Pinner or DAGService
- Dynamic configuration of libp2p ...

Here's an example plugin that overrides the default Pinner with a custom one:

> ```go
> func (p *PinnerPlugin) Options(opts []fx.Option) ([]fx.Option, error) {
> 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
> 	defer cancel()
> 
> 	cfg, err := config.LoadDefaultConfig(context.Background())
> 	if err != nil {
> 		return nil, fmt.Errorf("loading AWS config: %w", err)
> 	}
> 
> 	logger, err := zap.NewProduction()
> 	if err != nil {
> 		return nil, fmt.Errorf("constructing logger: %w", err)
> 	}
> 
> 	smClient := secretsmanager.NewFromConfig(cfg)
> 
> 	pinsDB, err := db.NewPinsDB(ctx, logger.Sugar(), smClient)
> 	if err != nil {
> 		return nil, fmt.Errorf("constructing PinsDB: %w", err)
> 	}
> 
> 	pinner := &pinner.Pinner{PinsDB: pinsDB}
> 
> 	return append(opts, fx.Replace(fx.Annotate(pinner, fx.As(new(pin.Pinner))))), nil
> }
> 
> ```

Reviewer 2: ```suggestion
```go
func (p *PinnerPlugin) Options(info core.FXNodeInfo) ([]fx.Option, error) {
	pinner := mypinner.New()    
	return append(info.FXOptions, fx.Replace(fx.Annotate(pinner, fx.As(new(pin.Pinner))))), nil
}

```
```
